wellness.sql

```diff
@@ -104,18 +104,137 @@ aggregate_wellness_survey_data AS (
+--Sum students that have a red or blue covitality color at some point during 2020-21AY
+sum_of_blue_red_covi AS (
+    SELECT
+        site_short,
+        SUM(wellness_blue_red_denom) AS sum_of_blue_red_covi_for_avg #students with blue/red Covitality scorecard colors for denominator
     FROM
-        students_that_completed_covi AS A
-        LEFT JOIN aggregate_wellness_survey_data AS B ON A.site_short = B.site_short
-)
+        gather_red_blue_covi_at
+    GROUP BY
+        site_short
+),
+--gather Wellness sessions attended during 2020-21
+gather_wellness_attendance_data AS (
+    SELECT
+        SUM(attendance_numerator_c) AS sum_attended_wellness_sessions,
+        site_short
+    FROM
+        `data-warehouse-289815.salesforce_clean.class_template` CT
+        LEFT JOIN `data-warehouse-289815.salesforce_clean.contact_at_template` CAT ON CAT.AT_Id = CT.Academic_Semester_c
+    WHERE
+        Attendance_Numerator_c > 0
+        AND department_c = 'Wellness'
+        AND dosage_types_c NOT LIKE '%NSO%'
+        AND AY_NAME = "AY 2020-21"
+        AND grade_c != '8th Grade'
+        AND Outcome_c != 'Cancelled'
+        AND college_track_status_c = '11A'
+        AND co_vitality_scorecard_color_c IN ('Blue', 'Red')
+    GROUP BY
+        site_short
+),
+--Prepare case note data for aggregation: red/blue covi as receiving 1:1 support by site. Will be used to add together with Wellness sessions attended
+--1 casenote = 1 session
+gather_case_note_data AS (
+    SELECT
+        CASE
+            WHEN id IS NOT NULL THEN 1
+            ELSE 0
+        END AS wellness_case_note_2020_21,
+        #wellness casenotes from 2020-21
+        id AS case_note_id,
+        #case note id
+        site_short
+    FROM
+        `data-warehouse-289815.salesforce_clean.contact_at_template` CAT
+        LEFT JOIN `data-warehouse-289815.salesforce.progress_note_c` CSE ON CAT.AT_Id = CSE.Academic_Semester_c
+    WHERE
+        Type_Counseling_c = TRUE
+        AND AY_name = 'AY 2020-21'
+        AND college_track_status_c = '11A'
+        AND co_vitality_scorecard_color_c IN ('Blue', 'Red')
+    GROUP BY
+        site_short,
+        id
+),
+--Sum case notes logged by site, for students with red or blue covitality scorecard color
+sum_of_case_notes AS (
+    SELECT
+        SUM(wellness_case_note_2020_21) as sum_of_wellness_case_notes,
+        site_short
+    FROM
+        gather_case_note_data
+    GROUP BY
+        site_short
+),
+--Add 1:1 case notes and sessions attended by student to average out later
+combine_sessions_and_case_notes AS (
+    SELECT
+        ATTNDCE.site_short,
+        SUM(
+            sum_attended_wellness_sessions + sum_of_wellness_case_notes
+        ) AS sum_wellness_support_received
+    FROM
+        gather_wellness_attendance_data AS ATTNDCE
+        LEFT JOIN sum_of_case_notes AS CSE ON ATTNDCE.site_short = CSE.site_short
+    GROUP BY
+        site_short
+),
+calculate_avg_wellness_services_per_blue_red_covi AS (
+    SELECT
```

Comment:
@isabelreyna, I might have to think about the math behind this, but you might be aggregating this data in the wrong order. 

it looks like you aggregated each component of this KPI by site, and then added them all up, and then averaged everything. In a sense you took an average of an average. 

The math might be same (I'd probably have to think about it) but I think it would be much easier to debug if something happened if you did everything on the student level and then aggregated at the end. 

So you have a row for every student and columns that indicate if they are red/blue, # of casenotes, # wellness sessions, etc. and then aggregate by site in that table. 

At the very least you could then very quickly untangle which students are showing up in the query instead of having everything aggregated at the beginning and making it hard to figure out which students did what.

Reply:
@jbakerr might need to talk through this, since I am unsure how it would be an avg of avg.  In simplest terms, what I **tried** to do is # sessions or case notes / red/blue students.
Add up all sessions attended for each red/blue student, then aggregate by site. Add up all case notes for red/blue students, then aggregate by site. Then divide by # of students.

I broke it down this way because I couldn't get aggregations to function the way I wanted.  There is probably an easier and cleaner way, but it ended up working out with tiny CTEs

Reply:
I had a similar reaction when reading through. But I think I just didn't understand the KPI. 

After reading through your response @isabelreyna I think it makes sense to me in that way. May help to chat through in tomorrow.

Reply:
@isabelreyna opportunity to revisit to refactor in July